app/partials/coinify-modal.pug

```diff
@@ -1,106 +1,60 @@
-    button.button-muted(ng-show="onStep('summary') && !error" ng-click="goTo('select-payment-medium')" translate="GO_BACK")
-    .flex-1.flex-end(ng-hide="status.loading")
-      button.button-muted.mrm(ng-click="$close() && goToOrderHistory()" translate="CLOSE" ng-show="onStep('trade-formatted') && bitcoinReceived")
-      button.button-muted.mrm(ng-click="close()" translate="CANCEL_ORDER" ng-hide="afterStep('summary') || onStep('summary') && error")
-      button.button-muted.mrm(ng-click="close() && goToOrderHistory()" translate="FINISH_LATER" ng-show="onStep('isx')")
-      div(ng-hide="afterStep('summary')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="nextStep()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('select-country')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="nextStep()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('email')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="signup()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('accept-terms')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="confirmOrContinue()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('select-payment-medium')")
-        button.button-primary(
-          ladda-translate="CONFIRM"
-          ng-click="buy()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled() || fiatFormInvalid || status.gettingQuote"
-          ng-show="onStep('summary') && !error")
-      div(ng-show="onStep('trade-in-review') || onStep('summary') && error || onStep('trade-formatted') && !bitcoinReceived")
-        button.button-primary(translate="CLOSE" ng-click="cancel() && goToOrderHistory()")
-      div(ng-show="onStep('trade-formatted') && bitcoinReceived")
-        button.button-primary(translate="SEE_BITCOIN" ng-hide="inMobileBuy" ng-click="cancel()" ui-sref="wallet.common.transactions")
-        button.button-primary(translate="SEE_BITCOIN" ng-show="inMobileBuy" ng-click="exitToNativeTx()")
+  
+  div(ng-if="onStep('email')" ng-controller="CoinifyEmailController" ng-include="'partials/coinify/email.pug'")
+  div(ng-if="onStep('accept-terms')" ng-controller="CoinifySignupController" ng-include="'partials/coinify/signup.pug'")
+  div(ng-if="onStep('select-payment-medium')" ng-controller="CoinifyMediumController" ng-include="'partials/coinify/medium.pug'")
+  div(ng-if="onStep('summary')" ng-controller="CoinifySummaryController" ng-include="'partials/coinify/summary.pug'")
```

Comment:
I like this idea, the reason I chose to use controllers was to have easy access to `$scope.vm.quote` or `$scope.vm.exchange`. In `accept-terms` for example, the parent scope needs to be refreshed before continuing, otherwise you get to summary with a fake quote. I'd rather set an updated property to `$scope.vm.property = property` than using directives, which always get messy. What is the downside of doing it this way?